contracts/voting-verifier/src/execute.rs

```diff
@@ -260,7 +264,23 @@ pub fn end_poll(deps: DepsMut, env: Env, poll_id: PollID) -> Result<Response, Co
         state::Poll::ConfirmWorkerSet(_) => end_poll_worker_set(deps, poll_id, &poll_result)?,
     };
 
+    // TODO: change rewards contract interface to accept a list of addresses to avoid creating multiple wasm messages
+    let rewards_msgs = poll_result
+        .consensus_participants
+        .iter()
+        .map(|address| WasmMsg::Execute {
+            contract_addr: config.rewards_contract.to_string(),
+            msg: to_binary(&rewards::msg::ExecuteMsg::RecordParticipation {
+                event_id: poll_id.into(),
+                worker_address: address.to_string(),
+            })
+            .expect("failed to serialize message for rewards contract"),
+            funds: vec![],
+        })
+        .collect::<Vec<_>>();
```

Comment:
```suggestion
        });
```
Don't even need to collect into a `Vec` cuz `add_messsages` takes `msgs: impl IntoIterator<Item = M>`